### PR TITLE
fix(type): fix incorrect ref unwrap

### LINKED ISSUE #257: Unexpected type inference when creating ref in function call
I'm getting unexpected type inference when creating a `ref` in a function call. The `ref('')` is interpreted as `ref<number>` which could definitely cause bugs (this actually happened to me as I was writing a unit test). The simple example below should provide all of the details but please let me know if there is anything else you need!

```typescript
const foo = (bar: Ref<number>) => {
  return bar;
};

// there are no TS errors in any of this code
// ref('') -> ref<unknown, string, number>(raw: string): Ref<number>
const result = foo(ref('')); // result -> Ref<number>
```

Thanks for your help!

### PR DESCRIPTION
**Pull request by antfu**
resolve #257

Align with vue-next
https://github.com/vuejs/vue-next/blob/c6217b4d46cbd3d9e250f2f84de4846d3484c42d/packages/reactivity/src/ref.ts#L30-L37

~I am not sure how these lines work in our current codebase~
```ts
// with init value: a = ref({ a: ref(0) })
// typeof a will be Ref<{ a: number }>
export function ref<S, T = unknown, R = HasDefined<S> extends true ? S : RefValue<T>>(
  raw: T
): Ref<R>;
```
~I kept it for now. But since it's not in vue-next, should we just remove it in order to have better type consistency with vue-next?~

**Edit**: Removed and it's exactly matched to vue-next now.

### CODE REVIEW

**Comment by pikax** [comment]
Once https://github.com/vuejs/composition-api/pull/311 is merged we should be able keep the `vue-next` types in sync easily

**Comment by antfu** [comment]
@pikax This seems to not be covered by #311. I have made it exactly matched to vue-next now.

File: package.json [hunk 1]
```diff
@@ -42,6 +42,7 @@
     "cross-env": "^7.0.2",
     "husky": "^4.2.5",
     "jest": "^26.0.1",
+    "jest-cli": "^26.0.1",
```
Comment (pikax) [inline package.json:45]: Is jest cli needed?

Comment (antfu) [reply package.json]: You are right, my mistake

**Pull request merged**